### PR TITLE
[api] fix duplicates in missing owner results

### PR DESCRIPTION
**Pull request by adrianschroeter**
caused by maintenance_release projects where the package name got not
shortened

### CODE REVIEW

**Comment by dmarcoux** [comment]
There's a typo in the commit message. duplic**a**tes

File: src/api/app/models/owner_search/missing.rb [hunk 1]
```diff
@@ -74,8 +78,12 @@ def defined_packages
     end
 
     def all_packages
-      ret = Package.where(project_id: @projects).pluck(:name)
-      ret.reject { |p| p =~ /\A_product:\w[-+\w.]*\z/ }
+      # package names from nonie maintenance projects
```
Comment (dmarcoux) [inline src/api/app/models/owner_search/missing.rb:81]: What is `nonie`?

**Comment by adrianschroeter** [comment]
clarified the comments hopefully

**Comment by dmarcoux** [comment]
Regarding the linters in the `checkout_code` check, the RuboCop offenses can be autocorrected with the command `bundle exec rake dev:lint:rubocop:auto_correct`.

**Comment by eduardoj** [comment]
I rebased the branch on master and fixed the Rubocop offenses with the `bundle exec rake dev:lint:rubocop:auto_correct` command.

**Pull request merged**